pkg/monitortests/node/legacynodemonitortests/exclusions.go

```diff
@@ -17,30 +16,52 @@ func isThisContainerRestartExcluded(locator string, exclusion Exclusion) bool {
 	// Our goal is to conquer these restarts.
 	// So we are sadly putting these as exceptions.
 	// If you discover a container restarting more than 3 times, it is a bug and you should investigate it.
-	exceptions := []string{
-		"container/metal3-static-ip-set",      // https://issues.redhat.com/browse/OCPBUGS-39314
-		"container/ingress-operator",          // https://issues.redhat.com/browse/OCPBUGS-39315
-		"container/networking-console-plugin", // https://issues.redhat.com/browse/OCPBUGS-39316
+	type exceptionVariants struct {
+		containerName      string
+		platformsToExclude string
+		topologyToExclude  string
 	}
-
-	// Upgrades seem to have a lot of failures.
-	// Let's exclude these for now generally.
-	if exclusion.upgradeJob {
-		return true
+	exceptions := []exceptionVariants{
+		{
+			// In this case, we found that we only saw failures for this container on bare metal.
+			// We did not find failures for vsphere where this is also run
+			// So if we start seeing failures on vsphere this would be a regression.
+			containerName:      "container/metal3-static-ip-set", // https://issues.redhat.com/browse/OCPBUGS-39314
+			platformsToExclude: "metal",
+		},
+		{
+			// ingress operator seems to only fail on the single topology.
+			// platform did not matter.
+			containerName:     "container/ingress-operator", // https://issues.redhat.com/browse/OCPBUGS-39315
+			topologyToExclude: "single",
```

Comment:
This one is a topology to *include* isn't it?

Reply:
I was thinking no. Based on what you suggested, we are seeing falures for ingress-operator single so I want to exclude single. 

I figured having one field where it is exclude and other than is include would cause me to cry in the long run so I decided against that.